crates/sparrow-physical/src/expr.rs

```diff
@@ -0,0 +1,91 @@
+use std::borrow::Cow;
+
+use arrow_schema::DataType;
+
+/// The identifier (index) of an expression.
+#[derive(Debug, serde::Serialize, serde::Deserialize, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(transparent)]
+#[serde(transparent)]
+pub struct ExprId(usize);
+
+impl From<usize> for ExprId {
+    fn from(value: usize) -> Self {
+        ExprId(value)
+    }
+}
+
+/// A physical expression which describes how a value should be computed.
+///
+/// Generally, each expression computes a column of values from zero or more
+/// input columns. Expressions appear in a variety of places within the steps
+/// that make up a physical plan.
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+
+pub struct Expr {
+    pub kind: ExprKind,
```

Comment:
Maybe add a one line comment here to explain what we mean with `kind`. I would throw the other names like `opcode` or `function`